app/src/main/java/cloudcity/GPSMonitor.java

```diff
@@ -211,20 +211,20 @@ public void requestCurrentLocation() {
         boolean lacksFineLocationPermission = ActivityCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED;
         boolean lacksCoarseLocationPermission = ActivityCompat.checkSelfPermission(context, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED;
         if (lacksFineLocationPermission || lacksCoarseLocationPermission) {
-            Log.wtf(TAG, "We lack ACCESS_FINE_LOCATION or ACCESS_COARSE_LOCATION permissions!");
+            CloudCityLogger.wtf(TAG, "We lack ACCESS_FINE_LOCATION or ACCESS_COARSE_LOCATION permissions!");
             return;
         }
 
         Executor executor = Executors.newSingleThreadExecutor();
         Consumer<Location> locationConsumer = new Consumer<Location>() {
             @Override
             public void accept(Location location) {
-                Log.d(TAG, "Current location is: " + location);
+                CloudCityLogger.d(TAG, "Current location is: " + location);
                 lastLocation = location;
                 if (location != null) {
-                    Log.d(TAG, "Current location's LAT: " + location.getLatitude() + ", LNG: " + location.getLongitude());
+                    CloudCityLogger.d(TAG, "Current location's LAT: " + location.getLatitude() + ", LNG: " + location.getLongitude());
                 } else {
-                    Log.w(TAG, "Cannot get current location's (LAT,LNG) because location was NULL");
+                    CloudCityLogger.w(TAG, "Cannot get current location's (LAT,LNG) because location was NULL");
```

Comment:
it is a worse type of error to not have these permissions than the typical "something went wrong" type of errors, which is what `.e()` is being used for.

in our default usecase, not having permissions is something that should never happen. it could, but shouldn't.